packages/lib/src/vis/utils.ts

```diff
@@ -46,16 +66,27 @@ const adaptedLogTicksThreshold = scaleLinear({
   range: [0.8, 1.4],
 });
 
-export function createAxisScale(
+export function createScale(
   scaleType: VisScaleType,
-  config: VisxScaleConfig | ScaleGammaConfig
-): AxisScale {
-  if (isScaleType(scaleType)) {
-    return H5WEB_SCALES[scaleType].createScale(config);
+  config: ScaleConfig
+): Scale {
+  if (Array.isArray(scaleType)) {
+    const [, exponent] = scaleType;
+    return scaleGamma({ ...(config as ScaleGammaConfig), exponent });
   }
 
-  const [, exponent] = scaleType;
-  return H5WEB_SCALES[ScaleType.Gamma].createScale({ ...config, exponent });
```

Comment:
Typing is tricky to get right with this pattern (we used to type the `config` parameter with `any`...) – especially if we decide to introduce generics. The `switch` statement gets rid of this complexity by removing a layer of abstraction.